SV-COMP 22 fixes

FrNecas opened the pull request:
@peterschrammel this is a fix for SVcomp, the same change that we had in our fork was applied in the develop branch ( see https://github.com/diffblue/cbmc/pull/5660 ) so it is no longer necessary. Reverting for now, we can drop these commits if we later rebase to a newer version.

FrNecas committed:
Revert "Witnesses: add the 'creationtime' attribute"

This reverts commit 40046b2c64e410b08900baaae92f4a3f014f194b. A commit
achieving the same thing is already present in CBMC causing duplication
of the attribute.

FrNecas committed:
Correctly consider inlining when searching abort

If inlining is performed, the function_identifier parameter will always
be __CPROVER_start. However, the former function is still stored in the
goto instruction and can be sued for the instrumentation

Signed-off-by: František Nečas <frantisek.necas@protonmail.com>

FrNecas commented:
One more bug uncovered, memcleanup instrumentation was not done, the `function_identifier` param was always set to `__CPROVER_start` due to inlining.

viktormalik commented:
> One more bug uncovered, memcleanup instrumentation was not done, the `function_identifier` param was always set to `__CPROVER_start` due to inlining.

You should probably rename the PR.

FrNecas committed:
Do not ignore edges if cond is true

This was previously commented out but was dropped in the rebase by
accident (the code was moved and merge conflict not correctly resolved).
This is necessary for some nontermination witnesses to function
correctly.

Signed-off-by: František Nečas <frantisek.necas@protonmail.com>

FrNecas commented:
One more fix added related to nontermination witnesses.

FrNecas commented:
Rebased to the current state of target branch. Lets see the results of one more pre-run and we can probably merge this if everything goes well :)

viktormalik commented:
@peterschrammel this should be ready, could you merge it, please? Thanks!
